mungegithub/mungers/assign-unassign-handler.go

```diff
@@ -0,0 +1,181 @@
+	RegisterMungerOrDie(dh)
+}
+
+// Name is the name usable in --pr-mungers
+func (AssignUnassignHandler) Name() string { return "assign-unassign-handler" }
+
+// RequiredFeatures is a slice of 'features' that must be provided
+func (AssignUnassignHandler) RequiredFeatures() []string { return []string{} }
+
+// Initialize will initialize the munger
+func (AssignUnassignHandler) Initialize(config *github.Config, features *features.Features) error {
+	return nil
+}
+
+// EachLoop is called at the start of every munge loop
+func (AssignUnassignHandler) EachLoop() error { return nil }
+
+// AddFlags will add any request flags to the cobra `cmd`
+func (h AssignUnassignHandler) AddFlags(cmd *cobra.Command, config *github.Config) {
+}
+
+// Munge is the workhorse the will actually make updates to the PR
+func (h AssignUnassignHandler) Munge(obj *github.MungeObject) {
+	if !obj.IsPR() {
+		return
+	}
+
+	comments, err := obj.ListComments()
+	if err != nil {
+		glog.Errorf("unexpected error getting comments: %v", err)
+		return
+	}
+
+	fileList, err := obj.ListFiles()
+	if err != nil {
+		glog.Errorf("Could not list the files for PR %v: %v", obj.Issue.Number, err)
+		return
+	}
+
+	//get all the people that could potentially own the file based on the blunderbuss.go implementation
+	potentialOwners, _ := getPotentialOwners(obj, h.features, fileList)
+
+	toAssign, toUnassign := h.assignOrRemove(obj, comments, fileList, potentialOwners)
+
+	//assign and unassign reviewers as necessary
+	for _, username := range toAssign.List() {
+		obj.AssignPR(username)
+	}
+	obj.Unassign(toUnassign.List()...)
+}
+
+//assignOrRemove checks to see when someone comments "/assign" or "/unassign"
+// returns two sets.String
+// 1. github handles to be assigned
+// 2. github handles to be unassigned
+// [TODO] "/assign <github handle>" assigns to this person
+func (h *AssignUnassignHandler) assignOrRemove(obj *github.MungeObject, comments []*goGithub.IssueComment, fileList []*goGithub.CommitFile, potentialOwners weightMap) (toAssign, toUnassign sets.String) {
+	toAssign = sets.String{}
+	toUnassign = sets.String{}
+
+	assignComments := commentMatcher.FilterComments(comments, commentMatcher.CommandName(assignCommand))
+	unassignComments := commentMatcher.FilterComments(comments, commentMatcher.CommandName(unassignCommand))
+	invalidUsers := sets.String{}
+
+	//collect all the people that should be assigned
+	for _, cmt := range assignComments {
+		if !mungerutil.IsValidUser(cmt.User) {
+			continue
+		}
+		if isValidReviewer(potentialOwners, cmt.User) {
+			glog.Infof("Assigning %v to review PR#%v", *cmt.User.Login, obj.Issue.Number)
+			toAssign.Insert(*cmt.User.Login)
+		} else {
+			// build the set of people who asked to be assigned but aren't in reviewers
+			// use the @ as a prefix so github notifies them in advance
+			invalidUsers.Insert("@" + *cmt.User.Login)
+		}
+
+	}
+
+	//collect all the people that should be unassigned
+	for _, cmt := range unassignComments {
```

Comment:
That still doesn't work. That /unassign is potentially old. You could maybe remove the comment once it's been processed (that would simplify a lot I think)